xdg-ninja.sh

```diff
@@ -196,21 +245,21 @@ check_file() {
 # Reads files from programs/, calls check_file on each file specified for each program
 do_check_programs() {
     while IFS="
-" read -r name; read -r filename; read -r movable; read -r help; do
-        check_file "$name" "$filename" "$movable" "$help"
+" read -r name; read -r filename; read -r movable; read -r help; read -r json_file;  do
+        check_file "$name" "$filename" "$movable" "$help" "$json_file"
     done <<EOF
-$(jq 'inputs as $input | $input.files[] as $file | $input.name, $file.path, $file.movable, $file.help' "$(dirname "$0")"/programs/* | sed -e 's/^"//' -e 's/"$//')
+$(jq 'inputs as $input | $input.files[] as $file | $input.name, $file.path, $file.movable, $file.help, input_filename' "$(dirname "$0")"/programs/* | sed -e 's/^"//' -e 's/"$//')
```

Comment:
Also, what about joining json files into a single json for more convenient parsing (without jq or with it)?

Reply:
A `-q|--quite` option is also needed for more convenient parsing.

Reply:
`--quiet`? I don't get it

Reply:
Currently if you use the `--json` flag the program first shills glow (if uninstalled), declares it is checking the homedir, outputs the json, says it is done checking the homedir, then asks for you to contribute to the github. All of this is also very inconvenient for people who wish to parse the output.

Reply:
Do it. But don't forget about joining json files. With current file format it is easy: you output `,` after each file and surround output with `[` and `]`

Reply:
Oh ok, I thought I would have to store all the jsons into a file, then run that file through `jq` at the end of the program.

Reply:
> Oh ok, I thought I would have to store all the jsons into a file, then run that file through `jq` at the end of the program.

Nevermind, my solution won't work, if you need to output files, not programs (take a look at json files). Asking `jq` to provide jsons for files and then reprinting those as I suggested would work